### PR TITLE
fixed update to save unparse job config resolves #634

### LINKED ISSUE #634: Roles of Job vs Execution Context
This has been bothering me for a while and I know I've brought it up before but now I think I realize the reason the way we currently store and validate jobs is not quite right given that we also have this notion of the execution context.  The thing that bothers me is that a users job gets validated and expanded and stored as a "Job" and then copied over as an execution context.  I suggest that we change how this works and only ever store the users "raw" (unvalidated) job on the job object and then do the validation and expansion on the execution context.  Conceptually speaking the job represents the users unmodified request and the execution context represents the the expansion of the job and thing that actually gets executed.

The realization I had was that the validation/expansion step then more tightly couples the job to implementation details of a specific version of the code.  You could be left with jobs that don't run after upgrading teraslice or with defaults from earlier versions.  Whereas if the job had not been validated/expanded it would be more likely to run after code changes.  Now, I don't consider this a real driving reason to make this change, it is more of a hint or clue that the current behavior is not quite right.

I realize this complicates the validation at submission time but that is probably manageable.

### PR DESCRIPTION
**Pull request by jsnoble**
integration tests pass

### CODE REVIEW

**Comment by jsnoble** [comment]
This requires changes made in teraslice-client module

File: integration-tests/spec/cases/assets/simple.js [hunk 1]
```diff
@@ -44,41 +42,38 @@ module.exports = function simpleAssetTest() {
     }
 
     describe('Asset Tests', () => {
-        describe('After uploading an asset', () => {
```
Comment (godber) [inline integration-tests/spec/cases/assets/simple.js]: You didn't like my grouping of tests with additional nested subscribes?

Comment (jsnoble) [reply integration-tests/spec/cases/assets/simple.js]: it was against how we formatted all the rest of the tests, describe was intended to determine the 'class' of test, and 'it' was to test all things related to that class

Comment (godber) [reply integration-tests/spec/cases/assets/simple.js]: Ok, but sometime you should take another look at the Jasmine docs and read about BDD to see what I think it's trying to achieve.  I don't think it matters much one way or another though.

File: integration-tests/spec/fixtures/jobs/generator-asset.json [hunk 2]
```diff
@@ -3,6 +3,7 @@
     "slicers": 1,
     "lifecycle": "persistent",
     "workers": 3,
+    "assets": ["ex1"],
```
Comment (godber) [inline integration-tests/spec/fixtures/jobs/generator-asset.json]: Is there a specific reason you add this here?  In all of your versions of the test and my versions of the test the `.assets` property gets overwritten by `jobSpec.assets = [JSON.parse(result)._id];`.  I had deliberately omitted it from here to keep the fixture generic.  Though I guess it's confusing that the job requires an asset but omits it.

**Comment by kstaken** [comment]
@godber are your questions blockers on merging this?

**Comment by godber** [comment]
Nah, go ahead.

**Comment by kstaken** [comment]
Integration tests pass and job update behavior manually verified.

**Pull request merged**